### PR TITLE
[fuchsia] mouse-input test

### PR DESCRIPTION
**Pull request by lin-erik**
This PR introduces the `mouse-input` integration test from `fuchsia.git`

This integration test validates mouse input by asserting both the location and buttons of an event match expected.

### CODE REVIEW

File: shell/platform/fuchsia/flutter/tests/integration/mouse-input/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Running the Test
+
+Reference the Flutter integration test [documentation](https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/tests/integration/README.md) at `//flutter/shell/platform/fuchsia/flutter/tests/integration/README.md`
+
+## Playing around with `mouse-input-view`
+
+Build Fuchsia with `workstation_eng.qemu-x64`
+```shell
+fx set workstation_eng.qemu-x64 --with-base=//src/session/bin/session_manager && fx build
```
Comment (akbiggs) [inline shell/platform/fuchsia/flutter/tests/integration/mouse-input/README.md:29]: Is `--with-base=//src/session/bin/session_manager` necessary here now that we're no longer using tiles_session?

File: shell/platform/fuchsia/flutter/tests/integration/utils/portable_ui_test.cc [hunk 2]
```diff
@@ -185,6 +177,22 @@ void PortableUITest::RegisterTouchScreen() {
   FML_LOG(INFO) << "Touchscreen registered";
 }
 
+void PortableUITest::RegisterMouse() {
+  FML_LOG(INFO) << "Registering fake mouse";
+  input_registry_ = realm_->Connect<fuchsia::ui::test::input::Registry>();
+  input_registry_.set_error_handler(
+      [](auto) { FML_LOG(ERROR) << "Error from input helper"; });
```
Comment (akbiggs) [inline shell/platform/fuchsia/flutter/tests/integration/utils/portable_ui_test.cc:184]: Does the argument give us any info about what the error was? Should we log it too?

**Review by akbiggs — APPROVED** [review]
Awesome! @abrush21 @jaeheon fyi

**Comment by jaeheon** [comment]
Thanks for the heads up. Is this a straightforward copy of the test from fuchsia.git? If so, that's helpful to know (cuts down on review time). Perhaps just say so in the commit log.

**Comment by lin-erik** [comment]
> Thanks for the heads up. Is this a straightforward copy of the test from fuchsia.git? If so, that's helpful to know (cuts down on review time). Perhaps just say so in the commit log.

Yeah, this is a straight port from fuchsia.git. I'll add that information to the squash commit description once I merge the PR in!

File: shell/platform/fuchsia/flutter/tests/integration/mouse-input/README.md (hunk 1 above)
Comment (lin-erik) [reply shell/platform/fuchsia/flutter/tests/integration/mouse-input/README.md]: Removed this from the `mouse-input` and `touch-input` READMEs

File: shell/platform/fuchsia/flutter/tests/integration/utils/portable_ui_test.cc (hunk 2 above)
Comment (lin-erik) [reply shell/platform/fuchsia/flutter/tests/integration/utils/portable_ui_test.cc]: Added `zx_status_get_string` to the log string

**Pull request merged**